### ISSUE TITLE
Incorrect transformation of VolumeRequest

### ISSUE THREAD

**Issue by justinrubek** [tristanpemble/nix-nomad#1]
# summary

[`job.<name>.group.<name>.volume`](https://tristanpemble.github.io/nix-nomad/index.html#opt-job._name_.group._name_.volume) attrs are transformed to JSON as a list. The final output ends up looking something like:
```
 "Volumes": [
  {                                                                                                                                                              
    "AttachmentMode": "file-system",
    "Name": "name",
    "ReadOnly": false,
    "Source": "name",
    ...
  },
   ...
],
```

However Nomad doesn't expect this to be a list but rather indexed by the volume's names:
```
"Volumes": {
        "name": {
          "Name": "name",
          "ReadOnly": false,
          "Source": "name",
          ...
        },
        ...
},
```

# what I've tried

- I'm running Nomad 1.4.3. It doesn't appear the syntax for volume has changed since then, but I've tried updating the nomad api module in go to see if it changes the generated nix module. No changes
- I am able to get this to work by manually changing [the relevant nix config](https://github.com/tristanpemble/nix-nomad/blob/054adcbdd0a836ae1c20951b67ed549131fd2d70/modules/generated.nix#L2320) to use `mapAttrs` instead of `mapAttrsToList`, however I have not figured out what needs to change in the generator to match

**Comment by tristanpemble** [comment]
apologies for the indecipherable Go code, it needs some serious love 😰 .

thanks for reporting this! I'm surprised we hadn't run into it yet. I merged a fix in #4, though I haven't tested it. let me know how that works for you.

**Comment by justinrubek** [comment]
Thank you! The newest version is working now. I'd been running it with the nix patched manually and my jobs look identical after switching back
